### PR TITLE
[Modules] Uplifted ADF module to support managed Private endpoints

### LINKED ISSUE #1720: [Feature Request]: Azure Data Factory Module to support Managed Private Endpoints
### Description

Azure Data Factory module contains a child resource for creating managed private endpoints.
https://docs.microsoft.com/en-us/azure/templates/microsoft.datafactory/factories/managedvirtualnetworks/managedprivateendpoints?tabs=bicep

These endpoints can hook into existing PaaS Services that are exposed behind private endpoints.

### PR DESCRIPTION
**Pull request by ahmadabdalla**
Closes #1720 

# Description

- Added support for a new child module for managed private endpoints for ADF
- Updated the pester test that inspects a private endpoint resource type to exclude managed private endpoints. The reason is that we are searching for the word 'privateEndpoints' using a like statement, which can conflict with 'managedPrivateEndpoints', thus marking it as the same for API versions. 

| Pipeline |
| - |
|[DataFactory: Factories](https://github.com/Azure/ResourceModules/actions/workflows/ms.datafactory.factories.yml) |

# Type of Change

Please delete options that are not relevant.

- [ ] New feature (non-breaking change which adds functionality)

# Checklist

- [ ] I'm sure there are no other open Pull Requests for the same update/change
- [ ] My corresponding pipelines / checks run clean and green without any errors or warnings
- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (readme)
- [ ] I did format my code

### CODE REVIEW

File: modules/Microsoft.DataFactory/factories/managedVirtualNetwork/deploy.bicep [hunk 1]
```diff
@@ -4,6 +4,9 @@ param dataFactoryName string
 @description('Required. The name of the Managed Virtual Network.')
 param name string
 
+@description('Optional. Enable telemetry via the Customer Usage Attribution ID (GUID).')
```
Comment (MrMCake) [inline modules/Microsoft.DataFactory/factories/managedVirtualNetwork/deploy.bicep:7]: This desscription seems off

File: modules/Microsoft.DataFactory/factories/managedVirtualNetwork/deploy.bicep [hunk 2]
```diff
@@ -29,6 +32,19 @@ resource managedVirtualNetwork 'Microsoft.DataFactory/factories/managedVirtualNe
...
+module managedVirtualNetwork_managedPrivateEndpoint 'managedPrivateEndpoints/deploy.bicep' = [for (managedPrivateEndpoint, index) in managedPrivateEndpoints: {
+  name: '${deployment().name}-managedPrivateEndpoint-${index}'
+  params: {
+    dataFactoryName: dataFactoryName
+    managedVirtualNetworkName: name
+    name: managedPrivateEndpoint.name
+    fqdns: managedPrivateEndpoint.fqdns
+    groupId: managedPrivateEndpoint.groupId
+    privateLinkResourceId: managedPrivateEndpoint.privateLinkResourceId
+    enableDefaultTelemetry: enableDefaultTelemetry
```
Comment (MrMCake) [inline modules/Microsoft.DataFactory/factories/managedVirtualNetwork/deploy.bicep:44]: This should reference a variable `var enableReferencedModulesTelemetry = false`

File: modules/Microsoft.DataFactory/factories/managedVirtualNetwork/deploy.bicep (hunk 1 above)
Comment (ahmadabdalla) [reply modules/Microsoft.DataFactory/factories/managedVirtualNetwork/deploy.bicep]: good pickup. Fixed

File: modules/Microsoft.DataFactory/factories/managedVirtualNetwork/deploy.bicep (hunk 2 above)
Comment (ahmadabdalla) [reply modules/Microsoft.DataFactory/factories/managedVirtualNetwork/deploy.bicep]: Fixed

**Review by MrMCake — APPROVED** [review]

**Pull request merged**